homeassistant/components/scrape/config_flow.py

```diff
@@ -95,19 +96,30 @@
     vol.Optional(CONF_VALUE_TEMPLATE): TemplateSelector(),
     vol.Optional(CONF_DEVICE_CLASS): SelectSelector(
         SelectSelectorConfig(
-            options=[cls.value for cls in SensorDeviceClass],
+            options=[
+                SelectOptionDict(value=item.value, label=item.value)
+                for item in SensorDeviceClass
+            ],
+            custom_value=True,
```

Comment:
It looks like simply adding an empty option solves the situation.
I think it should be resolved in frontend - but maybe this works for now.

```suggestion
            options=[
                SelectOptionDict(value=item.value, label=item.value)
                for item in SensorDeviceClass
            ]
            + [SelectOptionDict(value="", label="")],
```

Reply:
> It looks like simply adding an empty option solves the situation. I think it should be resolved in frontend - but maybe this works for now.

Thanks, but just to check did you test this? In my testing, an empty entry is entirely disregarded (see my comment https://github.com/home-assistant/core/pull/84482#issuecomment-1366258367 ). This was of course my original idea as it seemed the most logical. The only option I could find is to pass _some_ value in, and then use that value to instead be interpreted as a deletion of the original key, otherwise it just retains the old value.

Reply:
I used "None" in another integration for a user to choose and then had a validation to remove the key if that choice was made.

Reply:
Sorry - you are correct. The frontend was kind of working correctly but not storing in the storage.
I do have it working now with the simple:
```python
    vol.Optional(CONF_DEVICE_CLASS): SelectSelector(
        SelectSelectorConfig(
            options=[""] + [cls.value for cls in SensorDeviceClass],
            mode=SelectSelectorMode.DROPDOWN,
        )
    ),
    vol.Optional(CONF_STATE_CLASS): SelectSelector(
        SelectSelectorConfig(
            options=[""] + [cls.value for cls in SensorStateClass],
            mode=SelectSelectorMode.DROPDOWN,
        )
    ),
    vol.Optional(CONF_UNIT_OF_MEASUREMENT): SelectSelector(
        SelectSelectorConfig(
            options=[cls.value for cls in UnitOfTemperature],
            custom_value=True,
            mode=SelectSelectorMode.DROPDOWN,
        )
    ),
```